<?xml version="1.0"?>

<launch>
	<arg name="tello_ip" default="192.168.10.1" />
	<arg name="tello_cmd_server_port"	default="8889" />
	<arg name="local_cmd_client_port"	default="8890" />
	<arg name="local_vid_server_port"	default="6038" />
	<arg name="camera_calibration"		default="$(find tello_driver)/cfg/960x720.yaml" />    
	<arg name="namespace"			default="tello" />

	<group ns="$(arg namespace)">

		<node pkg="tello_driver" name="tello_driver_node" type="tello_driver_node" output="screen">
			<param name="local_cmd_client_port" value="$(arg local_cmd_client_port)" />
			<param name="local_vid_server_port" value="$(arg local_vid_server_port)" />
			<param name="tello_ip" value="$(arg tello_ip)" />
			<param name="tello_cmd_server_port" value="$(arg tello_cmd_server_port)" />
			<param name="connect_timeout_sec" value="10.0" />
			<param name="stream_h264_video" value="false" />
			<param name="camera_calibration" value="$(arg camera_calibration)" />			
		</node>
	</group>
	
	<!-- image transport -->
	<node pkg="image_transport" name="image_raw" type="republish" args="raw in:=/tello/camera/image_raw raw out:=/image" />

	<!-- zbar nodes -->
	<node pkg="nodelet" type="nodelet" name="zbar_manager"  args="manager"/>
	<node pkg="zbar_ros" type="barcode_reader_node" name="barcode_reader">
        	<param name="throttle_repeated_barcodes" type="double" value="2.0"/>
	</node>
	
	<!-- vrpn client node -->
	<node pkg="vrpn_client_ros" type="vrpn_client_node" name="vrpn_client_node" output="screen">

		<param name="server"                    value="192.168.2.2" />
                <param name="port"                      value="3883" />

                <param name="update_frequency"          value="100.0" />
                <param name="frame_id"                  value="world" />

                <param name="use_server_time"           value="false" />
                <param name="broadcast_tf"              value="true" />

                <param name="refresh_tracker_frequency" value="1.0" />

        </node>

</launch>
